File: Linear Data Structures/Arrays/Practice Problems Using Arrays/Java/Remove Duplicates.java
import java.util.HashSet;
import java.util.Scanner;

public class Main {
    public static void removeDuplicates(int[] arr, int n) {
        HashSet<Integer> set = new HashSet<>();
        int index = 0;
        for (int i = 0; i < n; i++) {
            if (!set.contains(arr[i])) {
                set.add(arr[i]);
                arr[index++] = arr[i];
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();
        
        int[] arr = new int[n];
        System.out.println("Enter the elements: ");
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        removeDuplicates(arr, n);
        System.out.print("Array after removing duplicates: ");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}

File: Linear Data Structures/Arrays/Practice Problems Using Arrays/C++/Remove Duplicates.cpp
#include <iostream>
using namespace std;

void removeDuplicates(int arr[], int& n) {
    int newSize = 0;
    for (int i = 0; i < n; i++) {
        int j;
        for (j = 0; j < newSize; j++) {
            if (arr[i] == arr[j]) {
                break;
            }
        }
        if (j == newSize) {
            arr[newSize++] = arr[i];
        }
    }
    n = newSize;
}

int main() {
    int n;
    cout << "Enter the number of elements: ";
    cin >> n;
    
    int arr[n];
    cout << "Enter the elements: ";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    removeDuplicates(arr, n);
    cout << "Array after removing duplicates: ";
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    return 0;
}
